src/app/forms/validators/custom-async-validators.ts

```diff
@@ -24,4 +28,15 @@ export class CustomAsyncValidators {
         })
       );
   }
+
+  static updateTestStationDetails(store: Store<State>): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<null> => {
+      store
+        .pipe(select(getTestStationFromProperty((control as CustomFormControl).meta.name as keyof TestStation, control.value)), take(1))
+        .subscribe(stations => {
+          stations && store.dispatch(updateTestStation({ payload: stations }));
+        });
```

Comment:
You were thinking of where to use `tap()`, right? 🙂
```suggestion
      const propertyName = (control as CustomFormControl).meta.name as keyof TestStation;
      
      store
        .pipe(
          select(getTestStationFromProperty(propertyName, control.value)),
          take(1),
          tap(stations => stations && store.dispatch(updateTestStation({ payload: stations })))
        )
        .subscribe();
```

Reply:
I did think about using the tap, but decided against it and just put it in the subscribe instead

Reply:
Don't you think it looks cleaner within `tap()`? It's like an effect of the store - do go and trigger something else after getting the value of the observable.

Reply:
feels like that would be using an operator for the sake of it. You have to subscribe anyway so might as well use it.